sparkmagic/sparkmagic/livyclientlib/livysession.py

```diff
@@ -153,31 +178,45 @@ def start(self):
             html = get_sessions_info_html([self], self.id)
             self.ipython_display.html(html)
 
-            command = Command("spark")
-            (success, out, mimetype) = command.execute(self)
+            self.set_session_context()
+            self.execute_profile()
 
-            if success:
-                self.ipython_display.writeln(u"SparkSession available as 'spark'.")
-                self.sql_context_variable_name = "spark"
-            else:
-                command = Command("sqlContext")
-                (success, out, mimetype) = command.execute(self)
-                if success:
-                    self.ipython_display.writeln(u"SparkContext available as 'sc'.")
-                    if ("hive" in out.lower()):
-                        self.ipython_display.writeln(u"HiveContext available as 'sqlContext'.")
-                    else:
-                        self.ipython_display.writeln(u"SqlContext available as 'sqlContext'.")
-                    self.sql_context_variable_name = "sqlContext"
-                else:
-                    raise SqlContextNotFoundException(u"Neither SparkSession nor HiveContext/SqlContext is available.")
         except Exception as e:
             self._spark_events.emit_session_creation_end_event(self.guid, self.kind, self.id, self.status,
                                                                False, e.__class__.__name__, str(e))
             raise
         else:
             self._spark_events.emit_session_creation_end_event(self.guid, self.kind, self.id, self.status, True, "", "")
 
+    def set_session_context(self):
+        command = Command("spark")
+        (success, out, mimetype) = command.execute(self)
+        if success:
+            self.ipython_display.writeln(u"SparkSession available as 'spark'.")
+            self.sql_context_variable_name = "spark"
+        else:
+            command = Command("sqlContext")
+            (success, out, mimetype) = command.execute(self)
+            if success:
+                self.ipython_display.writeln(u"SparkContext available as 'sc'.")
+                if ("hive" in out.lower()):
+                    self.ipython_display.writeln(u"HiveContext available as 'sqlContext'.")
+                else:
+                    self.ipython_display.writeln(u"SqlContext available as 'sqlContext'.")
+                self.sql_context_variable_name = "sqlContext"
+            else:
+                raise SqlContextNotFoundException(u"Neither SparkSession nor HiveContext/SqlContext is available.")
+
+    def execute_profile(self):
+        spark_profile_name = get_spark_profile_name()
+        if os.path.isfile(spark_profile_name):
+            command = Command("{}".format(open(spark_profile_name).read()))
```

Comment:
@gfreezy This idea of "profiles" is interesting - any chance you can explain what these profiles are intended for and how you're using them?